cansat-stm32f446/src/blink.rs

```diff
@@ -0,0 +1,28 @@
+//! Blinking LED using  blocking `Delay` in  TIM5.
+#![deny(unsafe_code)]
+
+use crate::hal::{pac, prelude::*};
+use panic_halt as _; // panic handler
+
+pub fn blink_loop() {
+    if let (Some(dp), Some(cp)) = (
+        pac::Peripherals::take(),
+        cortex_m::peripheral::Peripherals::take(),
+    ) {
+        // Setting up LED - its pin PA5 on Nucleo446RE
+        let gpioa = dp.GPIOA.split();
+        let mut led = gpioa.pa5.into_push_pull_output();
+        // Set up the system clock at48MHz.
+        let rcc = dp.RCC.constrain();
+        let clocks = rcc.cfgr.sysclk(48.MHz()).freeze();
+
+        // Create a delay based on SysTick
+        let mut delay = cp.SYST.delay(&clocks);
```

Comment:
`blink_loop` tries to do 2 things at once: initialization and the actual loop itself. I think you should just move the initialization to `main` and pass ownership of `led` and `delay` into the `blink_loop` fn